### PR TITLE
Policy deployment

### LINKED ISSUE #470: kubernetes dns deploy fails when calico policy_enabled: true
Hey all,

I've deployed k8s with kismatic successfully and wanted to try out enabling calico network policies. I redeployed my bare-metal machines and ran the kismatic installer with cluster:networking:policy_enabled: true. The deployment completes up to deploying the dns pods, but fails when waiting for them to be ready. 

Checking the logs on the pods/containers, nothing is remarkable other than in the kubedns container, 10.2.0.3 being the master's ip, which is strange as i can reach the api from my browser:
```
I0405 17:16:59.956279       1 dns.go:174] DNS server not ready, retry in 500 milliseconds
I0405 17:17:00.456228       1 dns.go:174] DNS server not ready, retry in 500 milliseconds
E0405 17:17:00.456401       1 reflector.go:199] k8s.io/dns/vendor/k8s.io/client-go/tools/cache/reflector.go:94: Failed to list *v1.Service: Get https://10.2.0.3:6443/api/v1/services?resourceVersion=0: dial tcp 10.2.0.3:6443: i/o timeout
E0405 17:17:00.456439       1 reflector.go:199] k8s.io/dns/vendor/k8s.io/client-go/tools/cache/reflector.go:94: Failed to list *v1.Endpoints: Get https://10.2.0.3:6443/api/v1/endpoints?resourceVersion=0: dial tcp 10.2.0.3:6443: i/o timeout
I0405 17:17:00.956070       1 dns.go:174] DNS server not ready, retry in 500 milliseconds
I0405 17:17:01.456136       1 dns.go:174] DNS server not ready, retry in 500 milliseconds
```
The calico-node container running on the machine trying to run kubedns shows in it's logs:
```
2017-04-05 17:14:09.790 [INFO][123] calc_graph.go 244: Local endpoint updated id=WorkloadEndpoint(node=daralis, orchestrator=k8s, workload=kube-system.kube-dns-3895116667-jz4hz, name=eth0)
2017-04-05 17:14:09.790 [WARNING][123] active_rules_calculator.go 260: Profile not known or invalid, generating dummy profile that drops all traffic. profileID="k8s_ns.kube-system"
2017-04-05 17:14:09.791 [INFO][123] int_dataplane.go 466: Received *proto.ActiveProfileUpdate update from calculation graph msg=id:<name:"k8s_ns.kube-system" > profile:<inbound_rules:<action:"deny" rule_id:"ZQQ7Duk5sGpQ5vQi" > outbound_rules:<action:"deny" rule_id:"hmRRFH_D_5H_-aEQ" > >
```
but then shortly after it looks like it's creating/opening the route:
```
2017-04-05 17:14:09.793 [INFO][123] endpoint_mgr.go 393: Updating endpoint routes. id=proto.WorkloadEndpointID{OrchestratorId:"k8s", WorkloadId:"kube-system.kube-dns-3895116667-jz4hz", EndpointId:"eth0"}
2017-04-05 17:14:09.794 [INFO][123] endpoint_mgr.go 466: Workloads changed, updating dispatch chains.
2017-04-05 17:14:09.794 [INFO][123] table.go 400: Queing deletion of chain. chainName="cali-from-wl-dispatch" ipVersion=0x6 table="filter"
2017-04-05 17:14:09.794 [INFO][123] table.go 627: Invalidating dataplane cache ipVersion=0x6 reason="chain removal" table="filter"
2017-04-05 17:14:09.794 [INFO][123] table.go 400: Queing deletion of chain. chainName="cali-to-wl-dispatch" ipVersion=0x6 table="filter"
2017-04-05 17:14:09.794 [INFO][123] table.go 627: Invalidating dataplane cache ipVersion=0x6 reason="chain removal" table="filter"
2017-04-05 17:14:09.794 [INFO][123] table.go 376: Queueing update of chain. chainName="cali-from-wl-dispatch" ipVersion=0x6 table="filter"
2017-04-05 17:14:09.794 [INFO][123] table.go 627: Invalidating dataplane cache ipVersion=0x6 reason="chain update" table="filter"
2017-04-05 17:14:09.794 [INFO][123] table.go 376: Queueing update of chain. chainName="cali-to-wl-dispatch" ipVersion=0x6 table="filter"
2017-04-05 17:14:09.794 [INFO][123] table.go 627: Invalidating dataplane cache ipVersion=0x6 reason="chain update" table="filter"
2017-04-05 17:14:09.794 [INFO][123] endpoint_mgr.go 692: Applying /proc/sys configuration to interface. ifaceName="cali94f4eef6e0e"
2017-04-05 17:14:09.794 [INFO][123] endpoint_mgr.go 313: Re-evaluated workload endpoint status adminUp=true failed=false known=true operUp=true status="up" workloadEndpointID=proto.WorkloadEndpointID{OrchestratorId:"k8s", WorkloadId:"kube-system.kube-dns-3895116667-jz4hz", EndpointId:"eth0"}
2017-04-05 17:14:09.794 [INFO][123] status_combiner.go 58: Storing endpoint status update ipVersion=0x6 status="up" workload=proto.WorkloadEndpointID{OrchestratorId:"k8s", WorkloadId:"kube-system.kube-dns-3895116667-jz4hz", EndpointId:"eth0"}
2017-04-05 17:14:09.794 [INFO][123] table.go 416: Loading current iptables state and checking it is correct. ipVersion=0x4 table="filter"
2017-04-05 17:14:09.800 [INFO][123] table.go 416: Loading current iptables state and checking it is correct. ipVersion=0x6 table="filter"
2017-04-05 17:14:09.807 [INFO][123] route_table.go 293: Syncing routes: adding new route. ifaceName="cali94f4eef6e0e" ipVersion=0x4 targetCIDR=172.16.179.66/32
2017-04-05 17:14:09.808 [INFO][123] status_combiner.go 81: Endpoint up for at least one IP version id=proto.WorkloadEndpointID{OrchestratorId:"k8s", WorkloadId:"kube-system.kube-dns-3895116667-jz4hz", EndpointId:"eth0"} ipVersion=0x4 status="up"
2017-04-05 17:14:09.808 [INFO][123] status_combiner.go 98: Reporting combined status. id=proto.WorkloadEndpointID{OrchestratorId:"k8s", WorkloadId:"kube-system.kube-dns-3895116667-jz4hz", EndpointId:"eth0"} status="up"
```

Do I need some other configuration or am I misunderstanding how to set this up?

### PR DESCRIPTION
**Pull request by dkoshkin**
Use a deployment for calico network policy.
Fix #470 

TODO wait for prod packages

### CODE REVIEW

File: ansible/_addon-network-policy.yaml [hunk 1]
```diff
@@ -8,3 +8,9 @@
 
     roles:
       - addon-network-policy
+
+    post_tasks:
+      - name: remove old calico-policy replication controller if exists
+        command: kubectl delete rc calico-policy -n kube-system
```
Comment (alexbrand) [inline ansible/_addon-network-policy.yaml]: I think we can use `--ignore-not-found` here?

File: ansible/roles/addon-network-policy/templates/policy-controller.yaml [hunk 2]
```diff
@@ -1,25 +1,29 @@
...
   template:
     metadata:
       name: calico-policy-controller
       namespace: kube-system
       labels:
-        kubernetes.io/cluster-service: "true"
-        k8s-app: calico-policy
+        tier: control-plane
+        k8s-app: calico-policy-controller
+        kismaticVersion: "{{ kismatic_short_version }}"
```
Comment (alexbrand) [inline ansible/roles/addon-network-policy/templates/policy-controller.yaml]: Should the ket version be at the deployment level?

File: ansible/roles/packages-kubernetes/tasks/main.yaml [hunk 3]
```diff
@@ -20,7 +20,7 @@
   # DEB
   - name: install kubelet deb package
     apt:
-      name: kubelet={{ kismatic_apt_version }}
+      name: kubelet={{ kismatic_deb_version }}
```
Comment (alexbrand) [inline ansible/roles/packages-kubernetes/tasks/main.yaml]: I guess we got lucky with these?

File: integration/install.go [hunk 4]
```diff
@@ -38,6 +38,7 @@ type installOptions struct {
 	modifyHostsFiles            bool
 	useDirectLVM                bool
 	serviceCIDR                 string
+	enableNetowrkPolicy         bool
```
Comment (alexbrand) [inline integration/install.go]: typo

File: integration/network-policy.go [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+	}
+
+	By("testing connection with policy disabled")
+	if err := testPodAccess(node, sshKey, 5); err != nil {
+		return fmt.Errorf("could not connect to pod: %v", err)
+	}
+
+	By("enabling global network policy on the default namespace")
+	if err := retry.WithBackoff(func() error {
+		return runViaSSH([]string{"sudo kubectl annotate ns default \"net.beta.kubernetes.io/network-policy={\\\"ingress\\\": {\\\"isolation\\\": \\\"DefaultDeny\\\"}}\" --overwrite"}, []NodeDeets{node}, sshKey, 1*time.Minute)
```
Comment (alexbrand) [inline integration/network-policy.go]: Might make sense to use backticks for this string.

File: integration/network-policy.go [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+	}, 3); err != nil {
+		return fmt.Errorf("could not set deny policy: %v", err)
+	}
+	// always try to disbale global policy
+	defer disableGlobalPolicy(node, sshKey)
```
Comment (alexbrand) [inline integration/network-policy.go]: Can a failure in disabling policy affect other tests?

File: integration/network-policy.go [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+	By("testing connection with global policy enable")
+	if err := testPodAccess(node, sshKey, 1); err == nil {
```
Comment (alexbrand) [inline integration/network-policy.go]: Is there a more meaningful test that we could perform here? It seems like we could get false-positives if there's a networking blip, or if the request fails for any other reason.

File: ansible/roles/packages-kubernetes/tasks/main.yaml (hunk 3 above)
Comment (dkoshkin) [inline ansible/roles/packages-kubernetes/tasks/main.yaml]: I only renamed the var, it was the same everywhere just wanted to make the name more consistent

**Comment by alexbrand** [comment]
LGTM!

**Comment by alexbrand** [comment]
This PR also fixes #515 by not mounting the `cni-net-dir` so that we can control when and how the CNI config is written out. In other words, we don't rely on Calico's script to configure CNI.

**Pull request merged**